File: Problems/No.205_Isomorphic_Strings/isomorphic_strings.cpp
class Solution {
public:
    bool isIsomorphic(string s, string t) {
        
        // parameters
        std::unordered_map<char, char> check_st, check_ts;

        // start computing
        if (s.length() != t.length()) return false;
        else {

            for (int i=0; i<s.length(); i++) {
                
                if (check_st[s[i]] && (check_st[s[i]] != t[i])) return false;
                if (check_ts[t[i]] && (check_ts[t[i]] != s[i])) return false;
                check_st[s[i]] = t[i];
                check_ts[t[i]] = s[i];

            }

        }

        return true;

    }
};

File: Problems/No.205_Isomorphic_Strings/isomorphic_strings.py
class Solution(object):
    def isIsomorphic(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """

        # start computing
        if (len(s) != len(t)):
            return False
        else:
            for i in range(len(s)):
                IDs = s.index(s[i])
                IDt = t.index(t[i])
                if (IDs != IDt):
                    return False
                else:
                    continue
            
        return True
